File: Selenium/Activities/Activity15.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity15 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/dynamic-attributes");
        System.out.println("Title of the page: " +driver.getTitle());
        // Find the username and password input fields and type in the credentials
        driver.findElement(By.xpath("//input[starts-with(@class,'username-')]")).sendKeys("admin");
        driver.findElement(By.xpath("//input[starts-with(@class,'password-')]")).sendKeys("password");
        driver.findElement(By.cssSelector("button.ui.button:nth-child(4)")).click();
        // Wait for login message to appear and print the login message
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        WebElement confirmMsg = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("action-confirmation")));
        System.out.println("Login message: " +confirmMsg.getText());
        // Close the browser
        driver.quit();
    }
}

File: Selenium/Activities/Activity15.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the browser to the URL
    driver.get("https://v1.training-support.net/selenium/dynamic-attributes")
    print("Title of the page is: ", driver.title)
    # Find the username and password input fields and type in the credentials
    driver.find_element(By.XPATH, "//input[starts-with(@class,'username-')]").send_keys("admin")
    driver.find_element(By.XPATH, "//input[starts-with(@class,'password-')]").send_keys("password")
    driver.find_element(By.CSS_SELECTOR, "button.ui.button:nth-child(4)").click()
    # Wait for login message to appear and print the login message
    wait = WebDriverWait(driver, 10)
    confirmMsg = wait.until(expected_conditions.visibility_of_element_located((By.ID, "action-confirmation")))
    print("Login Message: ", confirmMsg.text)
